### PR TITLE
KT-30322 Memory leak in CompilationSourceSetUtil

### PR DESCRIPTION
**Pull request by morj**

### CODE REVIEW

File: libraries/tools/kotlin-gradle-plugin-api/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinSourceSet.kt [hunk 1]
```diff
@@ -10,6 +10,7 @@ import org.gradle.api.Named
 import org.gradle.api.file.SourceDirectorySet
 
 interface KotlinSourceSet : Named, HasKotlinDependencies {
+    val id: Long
```
Comment (h0tk3y) [inline libraries/tools/kotlin-gradle-plugin-api/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinSourceSet.kt]: In a single project, source set names are unique. Using the `name` as a key would be sufficient. Why a separate `id` is needed?

Comment (morj) [reply libraries/tools/kotlin-gradle-plugin-api/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinSourceSet.kt]: Well, I am not a domain expert, but I can easily change this to maps of String.

Comment (morj) [reply libraries/tools/kotlin-gradle-plugin-api/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinSourceSet.kt]: Fixed

**Comment by h0tk3y** [comment]
Thanks!

**Pull request merged**